Xamarin.Forms.Platform.Android/AppCompat/TabbedPageRenderer.cs

```diff
@@ -400,51 +561,123 @@ protected virtual void SetTabIcon(TabLayout.Tab tab, FileImageSource icon)
-					_tabLayout.SetBackgroundColor(tintColor.ToAndroid());
+					if (tintColor.IsDefault && _backgroundDrawable != null)
+						_tabLayout.SetBackground(_backgroundDrawable);
+					else if (!tintColor.IsDefault)
+					{
+						if (_backgroundDrawable == null)
+							_backgroundDrawable = _tabLayout.Background;
+						_tabLayout.SetBackgroundColor(tintColor.ToAndroid());
+					}
 				}
 			}
 		}
 
 		void UpdateBarTextColor()
 		{
-			if (_disposed || _tabLayout == null)
-				return;
+			if (Element.OnThisPlatform().UseBottomNavigation())
+			{
+				if (_disposed || _relativeLayout == null || _bottomNavigationView == null)
+					return;
+
+				int currentColor = _bottomNavigationView.ItemTextColor.DefaultColor;
+
+        if (!_defaultColor.HasValue)
+					_defaultColor = currentColor;
+
+				Color newTextColor = Element.BarTextColor;
+				int newTextColorArgb = newTextColor.ToAndroid().ToArgb();
+				var color = newTextColor.ToAndroid();
+				color.A = 128;
+				int newTextColorArgbFade = color.ToArgb();
+
+				int[] sChecked = { global::Android.Resource.Attribute.StateChecked };
+				int[] sDefault = { };
+
+				if (!newTextColor.IsDefault && currentColor != newTextColorArgb)
+				{
+					_bottomNavigationView.ItemTextColor = new ColorStateList(new[] { sChecked, sDefault }, new[] { newTextColorArgb, color });
+					_bottomNavigationView.ItemIconTintList = new ColorStateList(new[] { sChecked, sDefault }, new[] { newTextColorArgb, color });
+
+				}
+			}
+			else
+			{
+				if (_disposed || _tabLayout == null)
+					return;
+
+				int currentColor = _tabLayout.TabTextColors.DefaultColor;
+
+				if (!_defaultColor.HasValue)
+					_defaultColor = currentColor;
 
-			int currentColor = _tabLayout.TabTextColors.DefaultColor;
+				Color newTextColor = Element.BarTextColor;
+				int newTextColorArgb = newTextColor.ToAndroid().ToArgb();
 
-			if (!_defaultColor.HasValue)
-				_defaultColor = currentColor;
+				if (!newTextColor.IsDefault && currentColor != newTextColorArgb)
+					_tabLayout.SetTabTextColors(newTextColorArgb, newTextColorArgb);
+				else if (newTextColor.IsDefault && _defaultColor.HasValue && currentColor != _defaultColor)
+					_tabLayout.SetTabTextColors(_defaultColor.Value, _defaultColor.Value);
+			}
+
+		}
 
-			Color newTextColor = Element.BarTextColor;
-			int newTextColorArgb = newTextColor.ToAndroid().ToArgb();
+		public bool OnNavigationItemSelected(IMenuItem item)
+		{
+			if (Element == null)
+				return false;
 
-			if (!newTextColor.IsDefault && currentColor != newTextColorArgb)
-				_tabLayout.SetTabTextColors(newTextColorArgb, newTextColorArgb);
-			else if (newTextColor.IsDefault && _defaultColor.HasValue && currentColor != _defaultColor)
-				_tabLayout.SetTabTextColors(_defaultColor.Value, _defaultColor.Value);
+			int selectedIndex = item.Order;
+			if (_bottomNavigationView.SelectedItemId != item.Order && Element.Children.Count > selectedIndex && selectedIndex >= 0)
```

Comment:
i think the first condition is meant to be __bottomNavigationView.SelectedItemId != item.ItemId_